test/unit/math/mix/meta/base_type_test.cpp

```diff
@@ -0,0 +1,65 @@
+#include <stan/math/mix.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/util.hpp>
+#include <complex>
+#include <vector>
+
+using stan::base_type;
+using stan::math::fvar;
+using stan::math::var;
+using d_t = double;
+using v_t = var;
+using fd_t = fvar<double>;
+using ffd_t = fvar<fd_t>;
+using fv_t = fvar<var>;
+using ffv_t = fvar<fv_t>;
+using md_t = Eigen::MatrixXd;
+using vd_t = Eigen::VectorXd;
+using rvd_t = Eigen::RowVectorXd;
+
+template <typename R, typename T>
+void expect_base() {
+  test::expect_same_type<R, typename stan::base_type<T>::type>();
```

Comment:
Any reason to test the type trait directly and not the alias?

Reply:
I'm testing both.  I was having trouble getting them to return the same thing in some cases initially, which is why I put the tests in.